src/js_schema.hpp

```diff
@@ -338,7 +339,17 @@ typename T::Object Schema<T>::object_for_property(ContextType ctx, const Propert
         Object::set_property(ctx, object, object_type_string, Value::from_string(ctx, property.object_type));
     }
     else if (is_array(property.type)) {
-        Object::set_property(ctx, object, object_type_string, Value::from_string(ctx, string_for_property_type(property.type & ~realm::PropertyType::Flags)));
+        auto tmp = string_for_property_type(property.type & ~realm::PropertyType::Flags);
+        if (strncmp(tmp, "string", 6) || strncmp(tmp, "int", 3) || strncmp(tmp, "bool", 4)
```

Comment:
Isn't this just a long-winded way of saying `if (property.type != realm::PropertyType::Object)`?